src/frontends/onnx/tests/onnx_import.in.cpp

```diff
@@ -6141,3 +6141,133 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_constant_of_shape_null_node) {
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, castlike_int32_to_float) {
+    auto function = onnx_import::import_onnx_model(file_util::path_join(CommonTestUtils::getExecutableDirectory(),
+                                                                        SERIALIZED_ZOO,
+                                                                        "onnx/castlike_int32_to_float64.onnx"));
+
+    auto test_case = test::TestCase(function, s_device);
+
+    test_case.add_input<int32_t>(Shape{1, 1, 2, 2}, std::vector<int32_t>{-1, 2, 3, 4});
+    test_case.add_input<float>(Shape{4}, {1, 2, 3, 4});
+    test_case.add_expected_output<float>(std::vector<float>{-1.0, 2.0, 3.0, 4.0});
+
+    test_case.run();
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, castlike_float64_to_int32) {
+    auto function = onnx_import::import_onnx_model(file_util::path_join(CommonTestUtils::getExecutableDirectory(),
+                                                                        SERIALIZED_ZOO,
+                                                                        "onnx/castlike_float64_to_int32.onnx"));
+
+    auto test_case = test::TestCase(function, s_device);
+
+    test_case.add_input<float>(Shape{1, 1, 2, 2}, std::vector<float>{-107374.9876543, -2.2, 3.3, 4.4});
+    test_case.add_input<int32_t>(Shape{4}, {1, 2, 3, 4});
+    test_case.add_expected_output<int32_t>(std::vector<int32_t>{-107374, -2, 3, 4});
+
+    test_case.run();
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, DISABLED_castlike_float32_to_bfloat16) {
+    auto function = onnx_import::import_onnx_model(file_util::path_join(CommonTestUtils::getExecutableDirectory(),
+                                                                        SERIALIZED_ZOO,
+                                                                        "onnx/castlike_float32_to_bfloat16.onnx"));
+
+    auto test_case = test::TestCase(function, s_device);
+
+    test_case.add_input<float>(Shape{3, 4}, std::vector<float>{121.5, 122.7, 3, 4, 5, 6, 7, 8.8, 9, 10, 11, 12});
+    test_case.add_input<bfloat16>(Shape{3, 4}, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12});
+    test_case.add_expected_output<bfloat16>(std::vector<bfloat16>{121.5, 122.7, 3, 4, 5, 6, 7, 8.8, 9, 10, 11, 12});
+
+    test_case.run();
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, DISABLED_castlike_bfloat16_to_float32) {
+    auto function = onnx_import::import_onnx_model(file_util::path_join(CommonTestUtils::getExecutableDirectory(),
+                                                                        SERIALIZED_ZOO,
+                                                                        "onnx/castlike_bfloat16_to_float32.onnx"));
+
+    auto test_case = test::TestCase(function, s_device);
+
+    test_case.add_input<bfloat16>(
+        Shape{3, 4},
+        std::vector<bfloat16>{121.5, 122.7, 3, 4, 5, 6, 7, 8.8, 9, 10, 11, 12});  // 122.7 is not 122.5, might be a bug,
+                                                                                  // create a ticket in JIRA for CPU
```

Comment:
do we have a ticket for it?

Reply:
I have to verify and create one if it doesn't exist yet. I've removed the comment though.